src/Commands/Stats.ts

```diff
@@ -0,0 +1,90 @@
+import { Command } from '../Infrastructure/System/Command';
+import { Message, Guild, GuildMember, MessageEmbed, Role, Collection } from 'discord.js';
+import moment = require('moment');
+
+/**
+ * ## Stats
+ * Display user Discord stats.
+ *
+ * Usage:
+ * `.stats`
+ * `.stats <discord ID/mention>`
+ *
+ * @category Commands: Misc
+ */
+export default class Stats extends Command {
+	public constructor() {
+		super({
+			name: 'Discord Stats',
+			command: [ 'stats' ],
+			description: 'Display user Discord stats.',
+			argumentDescriptions: [],
+			usage: '<prefix>stats <Argument>?',
+			roles: [],
+			commandsPerMinute: 5,
+			commandSurgeMax: 2.0,
+		});
+	}
+
+	public async execute(message: Message, args: string[]): Promise<void> {
+		const guild: Guild | undefined = this.client.guilds.cache.get('529310206639996928');
+		let guildMember: GuildMember | undefined | null = null;
+
+		if (guild && args.length > 0) {
+			if (guild.members.fetch(args[0])) {
+				guildMember = await guild.members.fetch(args[0]);
+			}
+		} else if (guild) {
+			guildMember = message.member;
+		} else {
+			return;
+		}
+
+		if (guildMember) {
+			const joinDiscord: string = `${ moment(guildMember.user.createdAt).format('lll') }\n` +
+				`*about ${ moment(guildMember.user.createdAt).fromNow() }*`;
+			const joinServer: string = `${ moment(guildMember.joinedAt).format('lll') }\n` +
+				`*about ${ moment(guildMember.joinedAt).fromNow() }*`;
+			const memberRoles: Collection<string, Role> = new Collection(
+				Array.from(guildMember.roles.cache.entries())
+			)
+				.sort((a: Role, b: Role) => b.position - a.position);
+
+			const roles: Role[] = [];
+			let rolesString: string = '*none*';
+			let status: string = guildMember.presence.status;
+
+			// iterate through user roles
+			memberRoles.forEach((el: Role) => {
+				if (el.name !== '@everyone') {
+					roles.push(el);
+				}
+			});
+
+			// make sure roles isn't empty
+			if (roles.length > 0) {
```

Comment:
We don't need to make sure this is empty, an empty array joined just returns an empty string.

Reply:
Oh I see, this is confusing because we interleave creating the variables with manipulating them. We should put all the memberRoles stuff together, and all the status stuff together. Also this might actually be clearer with a ternary:

```
let rolesString = roles.length > 0 ? roles.join(', ') : '*none*';
```